test/python/gemm/frost/test_matmul.py

```
@@ -2976,6 +2977,100 @@ def test_templates_route_version_gated_kwargs_through_the_guarded_wrappers():
+    so they run outside a kernel trace."""
+    import ast
+    import types
+
+    ns = dict(consts)
+    ns["cutlass"] = types.SimpleNamespace(const_expr=lambda x: x)
+    ns["nvvm"] = types.SimpleNamespace(Tcgen05MMACollectorOp=types.SimpleNamespace(FILL="FILL", USE="USE", LASTUSE="LASTUSE"))
+    tree = ast.parse(path.read_text())
+    found = {}
+    for node in tree.body:
+        if isinstance(node, ast.FunctionDef) and node.name in ("_a_collector_op", "_b_collector_op"):
+            exec(compile(ast.Module(body=[node], type_ignores=[]), str(path), "exec"), ns)
```

Ruff (0.16.2)

[error] Use of `exec` detected

(S102)